### PR TITLE
docker compose exec to return command exit code

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
Inspect exec after completion to return exitcode on `docker compose exec`

**Related issue**
close https://github.com/docker/compose-cli/issues/1607

### CODE REVIEW

File: cli/cmd/compose/exec.go [hunk 1]
```diff
@@ -108,5 +109,13 @@ func runExec(ctx context.Context, backend compose.Service, opts execOpts) error
...
 		execOpts.Reader = con
 	}
-	return backend.Exec(ctx, project, execOpts)
+	exitCode, err := backend.Exec(ctx, project, execOpts)
+	if exitCode != 0 {
+		errMsg := ""
+		if err != nil {
+			errMsg = err.Error()
+		}
+		return cli.StatusError{StatusCode: exitCode, Status: errMsg}
```
Comment (lorenrh) [inline cli/cmd/compose/exec.go:118]: can it be possible to have a non 0 exit code with no error 🤔 ?

File: local/compose/exec.go [hunk 2]
```diff
@@ -42,10 +42,10 @@ func (s *composeService) Exec(ctx context.Context, project *types.Project, opts
 		),
 	})
 	if err != nil {
-		return err
+		return 0, err
 	}
 	if len(containers) < 1 {
-		return fmt.Errorf("container %s not running", getContainerName(project.Name, service, opts.Index))
+		return 0, fmt.Errorf("container %s not running", getContainerName(project.Name, service, opts.Index))
```
Comment (lorenrh) [inline local/compose/exec.go:48]: do we really want this to be 0? 🤔

Comment (ndeloof) [reply local/compose/exec.go:48]: with another language I would have used `Either(ExitCode, Error)` but with golang we have to return both values for "everything went fine" and "oups" scenario, as a tuple.
and we can't use `nil` here until we make the signature to use `*int` which is not really better

Comment (ndeloof) [reply local/compose/exec.go:48]: as an additionl note, this is basically a copy/paste/adapt signature from https://github.com/docker/cli/blob/a32cd16160f1b41c1c4ae7bee4dac929d1484e59/cli/command/container/exec.go#L116
I'd like we can just rely on docker/cli for such interactions and don't have to duplicate code, maybe we can evolve in this direction making required changes to docker/cli once Compose 2.0 is a thing

**Review by lorenrh — APPROVED** [review]
LGTM 🌮

**Pull request merged**